__tests__/index.js

```diff
@@ -0,0 +1,113 @@
+/* global describe, expect, it */
+
+import fs from 'fs'
+import path from 'path'
+
+import accessibilityForGrid from '../lib/accessibility-for-grid'
+import generateDestinationData from '../lib/generate-destination-data'
+import getIsochrone from '../lib/get-isochrone'
+import getSurface from '../lib/get-surface'
+import {create as createGrid} from '../lib/grid'
+import {create as createOrigin} from '../lib/origin'
+import {create as createStopTreeCache} from '../lib/stop-tree-cache'
+
+const DATA = path.join(__dirname, '/data')
+const DEST_POINT = {x: 158, y: 140}
+const ORIGIN_POINT = {x: 125, y: 136}
+const which = 'MEDIAN'
+
+const jobsGrid = fs.readFileSync(`${DATA}/jobs.grid`).buffer
+const originData = fs.readFileSync(`${DATA}/origin.dat`).buffer
+const query = JSON.parse(fs.readFileSync(`${DATA}/query.json`, 'utf8'))
+const stopTrees = fs.readFileSync(`${DATA}/stop_trees.dat`).buffer
+
+const ctx = {}
+
+describe('Browsochrones', () => {
+  it('create stop tree cache', () => {
+    ctx.stopTreeCache = createStopTreeCache({
+      data: new Int32Array(stopTrees),
+      size: query.width * query.height
+    })
+    const {index} = ctx.stopTreeCache
+    expect(index.length).toMatchSnapshot()
+    expect(index[0]).toMatchSnapshot()
+    expect(index[index.length / 2]).toMatchSnapshot()
+    expect(index[index.length - 1]).toMatchSnapshot()
+  })
+
+  it('create grid', () => {
+    ctx.grid = createGrid(jobsGrid)
+    const {data, ...snapshot} = ctx.grid
+    expect(snapshot).toMatchSnapshot()
+  })
+
+  it('create origin', () => {
+    ctx.origin = createOrigin({
+      data: new Int32Array(originData),
+      point: ORIGIN_POINT
+    })
+    const {data, index, ...snapshot} = ctx.origin
+    expect(snapshot).toMatchSnapshot()
+  })
+
+  it('generate surface', () => {
+    ctx.surface = getSurface({
+      ...ctx,
+      query,
+      which
+    })
+    const {surface, ...snapshot} = ctx.surface
+    expect(filterSnapshot(snapshot)).toMatchSnapshot()
+  })
+
+  it('generate destination data', () => {
+    const snapshot = generateDestinationData({
+      ...ctx,
+      ...ctx.surface,
+      from: ORIGIN_POINT,
+      query,
+      to: DEST_POINT,
+      transitiveNetwork: query.transitiveData
+    })
+    expect(filterSnapshot(snapshot)).toMatchSnapshot()
+  })
+
+  it('get isochrone', () => {
+    const isochrone = getIsochrone({
+      ...query,
+      surface: ctx.surface.surface
+    })
+    expect(filterSnapshot(isochrone)).toMatchSnapshot()
+  })
+
+  it('get accessibility for grid', () => {
+    const access = accessibilityForGrid({
+      grid: ctx.grid,
+      query,
+      surface: ctx.surface.surface
+    })
+    expect(access).toMatchSnapshot()
+  })
+})
+
+function filterSnapshot (s, k) {
+  if (Array.isArray(s) || ArrayBuffer.isView(s)) {
+    return {
+      [`first${k ? `-${k}` : ''}`]: filterSnapshot(s[0]),
```

Comment:
I'm wondering if first and last is enough or if we should grab a few in the middle for added safety. I could see the first and last values both being zero (because for instance it's a grid of job locations and the first and last values are out in the woods somewhere) but there being something screwy in the middle (like all the values got multiplied by 10 or something, actually just fixed a bug in r5 where this was happening).